lib/nitcorn/reactor.nit

```diff
@@ -201,8 +201,15 @@ redef class Interfaces
 	redef fun add(e)
 	do
 		super
-		var config = vh.server_config
-		if config != null then sys.listen_on(e, config.factory)
+		var config = virtual_host.server_config
+		if config != null then register_and_listen(e, config)
+	end
+
+	# Indirection to `listen_on` and check if this targets all addresses
+	private fun register_and_listen(e: Interface, config: ServerConfig)
+	do
+		listen_on(e, config.factory)
+		if e.name == "0.0.0.0" or e.name == "::0" then config.default_virtual_host = virtual_host
```

Comment:
IPv6 addresses have multiple representations. For example, `:: == ::0 == 0::0 == ::0000`…